Code rearrangement

johenglisch opened the pull request:
Companion PR to https://github.com/glottobank/grambank-cldf/pull/125

xrotwang reviewed (approved):
Fine with me. The ultimate test might be recreating v1.0 of the CLDF dataset - exactly as it was.

johenglisch commented:
Okay, so I did this test run:

    #!/bin/sh
    set -e

    cd ~/repos/glottobank/grambank-cldf
    . ENV/bin/activate

    git -C raw/Grambank checkout v1.0.1

    git checkout master
    git -C ~/repos/grambank/pygrambank checkout master
    # disable zipping so diffs make sense
    sed '/dataset\.write/c\    dataset.write(**data)' \
        -i ~/repos/grambank/pygrambank/src/pygrambank/cldf.py

    # use the same glottolog that v1.0.1 was made from
    cldfbench makecldf cldfbench_grambank.py --glottolog-version v4.4
    git add cldf

    git -C ~/repos/grambank/pygrambank checkout src/pygrambank/cldf.py
    git checkout code-rearrangement
    git -C ~/repos/grambank/pygrambank checkout code-rearrangement

    cldfbench makecldf cldfbench_grambank.py --glottolog-version v4.4
    git status

Here's the relevant part of the `git status` output:

            modified:   CONTRIBUTORS.md
            modified:   cldf/StructureDataset-metadata.json
            modified:   cldf/contributors.csv
            modified:   cldf/requirements.txt
            modified:   raw/Grambank (modified content, untracked content)

Re contributors:  I had to copy over the `CONTRIBUTORS.md` from master because the code fails if the new *Should be included despite no coding* column isn't there.  Maybe I should make the code more lenient in that regard.

Other than that, it looks like this patch doesn't make any changes to the generated data.

btw, I also went back and compared the output to the actual v1.0.1:

    $ cp -r cldf cldf.bk
    $ git checkout cldf
    $ git checkout v1.0.1
    $ rm -r cldf
    $ cp -r cldf.bk cldf
    $ git status
    [...]
            modified:   CONTRIBUTORS.md
            modified:   cldf/StructureDataset-metadata.json
            modified:   cldf/contributors.csv
            modified:   cldf/languages.csv
            modified:   cldf/requirements.txt
            modified:   cldf/sources.bib
            modified:   cldf/values.csv
            modified:   raw/Grambank (modified content, untracked content)

Turns out it doesn't match Grambank 1.0 ‘exactly as it was’.  From skimming the diff it seems like `sourcelookup` code was already producing different output…

Also, we removed the `Coders` column from the language table because it can be infered from the value table.

Also also, the value table would be zipped if I hadn't turned that off for easier debugging (note to self, I forgot to reenable it).

xrotwang commented:
Sounds ok, I guess. We should make sure to only create CLDF data releases with released versions of pygrambank in the future - and possibly starting with v1.0.2?
